### DISCUSSION TITLE
Help Setting up LAN server

### DISCUSSION THREAD

**Discussion by Noob415** [azerothcore/azerothcore-wotlk#12104] [General]
first off sorry if this is the wrong place for this, i just discovered wow private servers yesterday so im super new to everything and am not sure where to post something like this. 

but im trying to setup a LAN server for me and my roomates. i found a repacked server that works perfectly for just me but i cant find what to change in order to get other people to connect. The repack i found was this:

https://www.guilded.gg/solofriendly/channels/6806b853-d39d-48e2-9b8d-3cc478767648/forums/111008508

everything from what i can see is set to 127.0.0.1 which is my machine. i dont know what/where to change that to be my LAN IP. in many different config files 127.0.0.1 is found and idk which one to change, all of them? ive tried many different combinations and i cant get anything to work. im sure this is a super simple fix and im just missing something but ive looked everywhere and the only information i can find is for other cores and using heidiSQL for external computers to connect and i just need LAN. any help would be appreciated, thank you!

**Comment by virgo77** [comment]
Hi!
Inside this repack, if you play on your own, so you don't need to change something, just only your client realmlist file (see below). But because you want to play with other players, you need to make some changes. And, as I read the authserver.conf and worlserver.conf files, the server should run whatever the LAN IP of your computer, because even in authserver.conf or worldserver.conf you can find a line called "BindIP=0.0.0.0"

So, if you want to play with other players, here is what you have to do:
- run the .BAT file, on the root folder of the server. That will launch the SQL instance and the game
- then, launch a SQL editor, like HeidiSQL or SQLyog Community
- connect to the SQL DB:
   IP: 127.0.0.1
   user: root
   pass: 123456
   port: 3310
   (all as said in the link you gave here)
- then, go to the 'acore_auth' db
- open the 'realmlist' table
- look at the Default Realm (the first line)
- you have to modify this row: **Address**. By default you will see 127.0.0.1. Change it with your computer local IP
- Don't change anything else
- save the Table, then the DB, and quit the SQL editor

Stop the worldserver (CTRL C in its window), authserver (same), and the DB

Go to your Windows Firewall and open these 2 ports (both UDP/TCP):
3724 (for authserver)
8085 (for worldserver)

Then, launch the .BAT file and wait your server is fully running

Now you need to create the accounts.
Go in the worldserver.exe window, and write these commands:
``account create USER PWD`` (where USER is the player account, and PWD the password)
``account set addon USER 2`` (to activate the WotLK expansion for 'USER')
Then repeat for each roommate

_**Optional:**
if you are the Game Master, so you can do these accounts creations directly from the game, in the chat window:
``.account create USER PWD``
``.account set addon USER 2``
(don't forget the dot at the beginning)_

Now, you have to tell your game how to connect to the server.
So both of all of you need to modify the 'realmlist.wtf' file, which is a text file, here:
`<game_client_directory>\data\en-US\realmlist.wtf` (if you are US Players)
or
`<game_client_directory>\data\en-GB\realmlist.wtf` (if you are British Players)
or
`<game_client_directory>\data\de-DE\realmlist.wtf` (if your are German Players)
etc.
es-SP=Spanish
fr-FR=France
fr-CA=Canada
ru-RU=Russia
And so go on.

First, make a copy of the original realmlist.wtf file.
Don't touch the copy, and open the original file with notepad or notepad++, delete all text and add this line:
``set realmlist LOCALIP.OF.YOUR.COMPUTER`` (ie: 192.168.1.xx)
save and close
Launch the game client and connect with the USER and PWD your created before
Normally, you will be able to connect, and create a first character, then play with it.


Nota Bene:
In your worldserver/authserver conf files, you will see these lines:
- authserver:
LoginDatabaseInfo = "127.0.0.1;3310;root;123456;acore_auth"
- worldserver:
LoginDatabaseInfo     = "127.0.0.1;3310;root;123456;acore_auth"
WorldDatabaseInfo     = "127.0.0.1;3310;root;123456;acore_world"
CharacterDatabaseInfo = "127.0.0.1;3310;root;123456;acore_characters"

As your DB Server is running on your computer, 127.0.0.1 is the same as localhost. It is your computer.
3310 is the DB Server connection port
root is the DB Server user account
123456 is the DB Server account password
acore_xxx is the name of the DB used by both authserver and worldserver
(where xxx is for 'auth' (player authentication) / characters (your characters (name, pet, etc.), achievements, etc.) / worldserver (the entire Azeroth universe (Kalimdor, East Kingdoms, Outland, Northrend (PNJs, maps, dungeons, raids, etc.))


Hope this can help you, and sorry for the late answer :)

Now, have fun!

**Reply by HerrDave** [reply]
I'm about to necro the Hell out of this thread, my apologies in advance

I've been using this repack:
https://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-emulator-servers/wow-emu-general-releases/976782-azerothcore-wrath-of-lich-king-single-player-repack.html
And I'm trying to open up the server to LAN, so my family can play co-op.

Following your instructions, I downloaded both HeidiSQL and SQLyog Community (in the event that one works and the other doesn't)
I launched the .BAT and opened the SQL editors - the fields were blank so I made a New Session (Heidi) and when I tried to open (Heidi) or Connect (SQLyog Community), it instantly gives me an Error

( SQLyog Community )
Error No. 1045
Access denied for user 'root'@'localhost' (using password: NO)
[ OK ]

( HeidiSQL )
Access denied for user 'root'@'localhost' (using password: NO)
[ OK ]

I've never used this program before, and even by blindly feeling around the options I can't seem to move forward.
What am I supposed to do here?

*Edit
I worked it out with this video -> https://www.youtube.com/watch?v=NdRK5WWfYp0&list=PLSqlqifGHQco53eb8R1yHS3Sy-UbJCceX&index=49

But WOW the setup is involved; hope this helps others!
